### PR TITLE
[MU4] Play panel infrastructure

### PR DESCRIPTION
**Pull request by RomanPudashkin**
Various improvements and fixes needed for the play panel

### CODE REVIEW

File: src/framework/uicomponents/qml/MuseScore/UiComponents/StyledPopupView.qml [hunk 1]
```diff
@@ -29,20 +30,31 @@ PopupView {
...
     readonly property int borderWidth: 1
 
+    function toggleOpened() {
+        if (isOpened) {
+            close()
+        } else {
+            open()
+        }
+    }
+
```
Comment (cbjeukendrup) [inline src/framework/uicomponents/qml/MuseScore/UiComponents/StyledPopupView.qml:46]: Couldn't this be done in "PopupView.cpp"? Or has it advantages to do it in QML?

File: src/framework/ui/view/iconcodes.h [hunk 2]
```diff
@@ -252,6 +253,7 @@ class IconCode
         NOTE_DOTTED_2 = 0xF395,
         NOTE_DOTTED_3 = 0xF396,
         NOTE_DOTTED_4 = 0xF397,
+        MIDI_INPUT = 0xEF98,
```
Comment (Eism) [inline src/framework/ui/view/iconcodes.h:256]: enums must be sorted by value

File: src/framework/uicomponents/qml/MuseScore/UiComponents/StyledPopupView.qml (hunk 1 above)
Comment (RomanPudashkin) [reply src/framework/uicomponents/qml/MuseScore/UiComponents/StyledPopupView.qml]: Yes, you are right. Moved to cpp

File: src/framework/ui/view/iconcodes.h (hunk 2 above)
Comment (RomanPudashkin) [reply src/framework/ui/view/iconcodes.h]: fixed, thanks

**Review by Eism — APPROVED** [review]

**Review by igorkorsukov — APPROVED** [review]

**Pull request merged**